File: spring-webflux-plain-java-example/src/test/java/de/cfranzen/examples/webflux/CustomerRepositoryTest.java
package de.cfranzen.examples.webflux;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.data.r2dbc.DataR2dbcTest;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;


@DataR2dbcTest
class CustomerRepositoryTest {

    @Autowired
    private CustomerRepository repository;

    @Test
    void testFindingByFirstAndLastName() {
        // given
        final var c1 = new Customer("Peter", "Mueller");
        final var c2 = new Customer("Peter", "Mueller");
        final var c3 = new Customer("Sabine", "Hochhaus");
        repository.saveAll(List.of(c1, c2, c3)).blockLast();

        // when
        final var result = repository.findByFirstNameAndLastName(c1.getFirstName(), c1.getLastName());

        // then
        assertThat(result.toStream().toList())
                .usingRecursiveFieldByFieldElementComparatorIgnoringFields("id")
                .containsExactly(c1, c2);
    }

}

File: spring-webflux-kotlin-example/src/test/kotlin/de/cfranzen/examples/webflux/kotlin/CustomerRepositoryTest.kt
package de.cfranzen.examples.webflux.kotlin

import kotlinx.coroutines.flow.collect
import kotlinx.coroutines.flow.toList
import kotlinx.coroutines.test.runTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.data.r2dbc.DataR2dbcTest

@DataR2dbcTest
internal class CustomerRepositoryTest(
    @Autowired private val repository: CustomerRepository
) {

    @Test
    fun testFindingByFirstAndLastName() = runTest {
        // given
        val c1 = Customer(firstName = "Peter", lastName = "Mueller")
        val c2 = Customer(firstName = "Peter", lastName = "Mueller")
        val c3 = Customer(firstName = "Sabine", lastName = "Hochhaus")
        repository.saveAll(listOf(c1, c2, c3)).collect()

        // when
        val result = repository.findByFirstNameAndLastName(
            firstName = c1.firstName,
            lastName = c1.lastName
        )

        // then
        assertThat(result.toList())
            .usingRecursiveFieldByFieldElementComparatorIgnoringFields("id")
            .containsExactly(c1, c2)
    }
}